platform/viewer/src/appExtensions/MeasurementsPanel/init.js

```diff
@@ -70,6 +77,45 @@ export default function init({
     );
   };
 
+  const showAnnotationDialog = toolName => {
+    if (!UIDialogService) {
+      console.warn('Unable to show dialog; no UI Dialog Service available.');
+      return;
+    }
+
+    switch (process.env.ANNOTATION_TYPE) {
+      case 'MAMA':
+        var content = SimpleDialog.AnnotationMAMADialog;
+        break;
+      case 'APA':
+        var content = SimpleDialog.AnnotationAPADialog;
+        break;
+      default:
+        return;
+    }
+
+    let dialogId = UIDialogService.create({
+      centralize: true,
+      isDraggable: false,
+      content: content,
+      useLastPosition: false,
+      showOverlay: true,
+      contentProps: {
+        onClose: () => UIDialogService.dismiss({ id: dialogId }),
+        onSubmit: value => {
+          console.log(value); // TODO when confirmed, send via HTTP POST
+          let csv = Object.keys(value)
+            .map(function(k) {
+              return value[k];
+            })
+            .join(',');
+          console.log(csv);
+          UIDialogService.dismiss({ id: dialogId });
+        },
+      },
+    });
+  };
```

Comment:
Ah! You did use the DialogService. Nice!
This is an interesting approach. It does make sense that some measurements would require different options to edit their associated data/labels. Maybe there's a more dynamic way to achieve this?

It looks like we have a default dialog, and then we could associate certain measurement types with registered edit interfaces? We could set those associations when configuring and registering the MeasurementsPanel extension?